### PR TITLE
[MNG-7463] Improve documentation about deprecation in Mojo

### PR DESCRIPTION
**Pull request by slawekjaranowski**
Following this checklist to help us incorporate your
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed
       for the change (usually before you start working on it).  Trivial changes like typos do not
       require a JIRA issue. Your pull request should address just this issue, without
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] SUMMARY`, where you replace `MNG-XXX`
       and `SUMMARY` with the appropriate JIRA issue. Best practice is to use the JIRA issue
       title in the pull request title and in the first line of the commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
Similar to https://github.com/apache/maven-site/pull/319

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java [hunk 1]
```diff
@@ -88,8 +88,8 @@
     private String executeLifecycle;
 
     /**
-     * Specify the version when the Mojo was deprecated to the API. Similar to Javadoc deprecated. This will trigger a
-     * warning when a user tries to configure a parameter marked as deprecated.
+     * Description with reason of Mojo deprecation. Similar to Javadoc deprecated.
```
Comment (michael-o) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java:91]: I'd do: Javadoc `{@code @deprecated}`

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java [hunk 2]
```diff
@@ -170,15 +170,15 @@ public void setLanguage( String language )
...
+     * @return Description with reason of a Mojo deprecation.
      */
     public String getDeprecated()
     {
         return deprecated;
     }
 
     /**
-     * @param deprecated <code>true</code> to deprecate the Mojo, <code>false</code> otherwise.
+     * @param deprecated Set description with reason of a Mojo deprecation.
```
Comment (michael-o) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java:181]: The parameter does not set, the method does. So the param desc should ge identical with `#getDeprecated()` return value.

File: maven-plugin-api/src/main/mdo/plugin.mdo [hunk 3]
```diff
@@ -290,8 +290,8 @@ under the License.
           <version>1.0.0</version>
           <type>String</type>
           <description>
-            Specify the version when the Mojo was deprecated to the API. Similar to Javadoc deprecated. This will
-            trigger a warning when a user tries to configure a parameter marked as deprecated.
+            Description with the reason of Mojo deprecation. Similar to Javadoc deprecated.
```
Comment (michael-o) [inline maven-plugin-api/src/main/mdo/plugin.mdo:293]: Same here

File: maven-plugin-api/src/main/mdo/plugin.mdo [hunk 4]
```diff
@@ -423,7 +423,7 @@ under the License.
           <version>1.0.0</version>
           <type>String</type>
           <description>
-            Specify the version when the parameter was deprecated to the API. Similar to Javadoc deprecated.
+            Description with the reason of parameter deprecation. Similar to Javadoc deprecated.
```
Comment (michael-o) [inline maven-plugin-api/src/main/mdo/plugin.mdo:426]: Same here

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java [hunk 5]
```diff
@@ -88,8 +88,8 @@
     private String executeLifecycle;
 
     /**
-     * Specify the version when the Mojo was deprecated to the API. Similar to Javadoc deprecated. This will trigger a
-     * warning when a user tries to configure a parameter marked as deprecated.
+     * Description with reason of Mojo deprecation. Similar to Javadoc {@code @Deprecated}.
```
Comment (michael-o) [inline maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/MojoDescriptor.java:91]: Attention: the JavaDoc tag is `@deprecated`. Not the annotation.

File: maven-plugin-api/src/main/mdo/plugin.mdo [hunk 6]
```diff
@@ -289,10 +289,10 @@ under the License.
           <name>deprecated</name>
           <version>1.0.0</version>
           <type>String</type>
-          <description>
-            Specify the version when the Mojo was deprecated to the API. Similar to Javadoc deprecated. This will
-            trigger a warning when a user tries to configure a parameter marked as deprecated.
-          </description>
+          <description><![CDATA[
+            Description with the reason of Mojo deprecation. Similar to Javadoc <code>@Deprecated</code>
```
Comment (michael-o) [inline maven-plugin-api/src/main/mdo/plugin.mdo:293]: same here

File: maven-plugin-api/src/main/mdo/plugin.mdo [hunk 7]
```diff
@@ -422,10 +422,10 @@ under the License.
           <name>deprecated</name>
           <version>1.0.0</version>
           <type>String</type>
-          <description>
-            Specify the version when the parameter was deprecated to the API. Similar to Javadoc deprecated.
+          <description><![CDATA[
+            Description with the reason of parameter deprecation. Similar to Javadoc <code>@Deprecated</code>
```
Comment (michael-o) [inline maven-plugin-api/src/main/mdo/plugin.mdo:426]: same here

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by slawekjaranowski** [comment]
> 
> @slawekjaranowski As you can see, even a non-functional change seems obvious one can still find inconsistencies, bugs, errors, etc.

Exactly, therefore I'm for RTC, 
as we see review can increase quality even of simple change

**Comment by slawekjaranowski** [comment]
oh and build also should pass on ci 😄

**Comment by michael-o** [comment]
Can this be applied to 3.9.x as well? Is the warning issued there too?

**Comment by slawekjaranowski** [comment]
> Can this be applied to 3.9.x as well? Is the warning issued there too?

Yes, I will do it, according to fix version in issue

**Comment by jira-importer** [comment]
Resolve #9133

**Pull request merged**